cmd/brew-report-issue.rb

```diff
@@ -13,19 +13,21 @@
   body = STDIN.read
 end
 
-@credential_helper = `git config --global credential.helper`.chomp
-
 strap_url = ENV["STRAP_URL"]
 strap_url ||= "https://strap.githubapp.com"
 
-if @credential_helper.empty?
+if `git config --global credential.helper`.chomp.empty?
   abort <<-EOS
 Error: your Git HTTP(S) credential helper is not set! Set it by running Strap:
 #{strap_url}
 EOS
 end
 
-@github_credentials=`printf "protocol=https\nhost=github.com\n" | git credential fill 2>/dev/null`
+def credential_helper(command, input)
+  `printf "#{input}" | git credential #{command} 2>/dev/null`
```

Comment:
This `printf` is potentially a problem both due to shell interpolation and `printf` interpolation. For the `fill` command, you feed a pre-defined string, but the output you get back (including the user's password) will get fed back to `approve` or `reject`. If the password has a double-quote, a backslash, or a percent, you're going to get bogus results.

You probably want to just pipe straight from ruby to `git credential`, and let the `\n` interpolation happen in the ruby string literal.